swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift

```diff
@@ -94,12 +129,36 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
 
   // TODO: Use a message format to allow requesting different types of data.
   // This currently assumes we're requesting resources.
-  override func handleAppMessage(_ hash: Data, completionHandler: ((Data?) -> Void)? = nil) {
-    adapter?.getResourcesIfVersionDifferentFrom(hash: hash) {
-      resourceListJSON in
-      completionHandler?(resourceListJSON?.data(using: .utf8))
+  override func handleAppMessage(_ message: Data, completionHandler: ((Data?) -> Void)? = nil) {
```

Comment:
This is the primary IPC callback from GUI -> Tunnel.

We overload the `message` parameter to determine what the GUI is asking the tunnel to do. This could be expanded in the future using JSON and more structured format, but I kept it simple for now as it only handle 2 commands.

Reply:
I believe so, it's a byte array